pkg/extension/manifest.go

```diff
@@ -65,6 +65,13 @@ func WithCustomAccessCheck(fn AccessCheckFunc) Option {
 	}
 }
 
+// WithCustomAuthPlugins specifies the custom authentication plugins available for the system.
+func WithCustomAuthPlugins(authPlugins map[string]*AuthPlugin) Option {
```

Comment:
Could we make the argument to `plugins []*AuthPlugin` to make it simple ? It's doesn't matter if the input plugins have dup names because we'll check it later